packages/playwright-core/src/client/browserContext.ts

```diff
@@ -90,6 +90,7 @@ export class BrowserContext extends ChannelOwner<channels.BrowserContextChannel>
     this._isChromium = this._browser?._name === 'chromium';
     this.tracing = Tracing.from(initializer.tracing);
     this.request = APIRequestContext.from(initializer.requestContext);
+    this.request._timeoutSettings = this._timeoutSettings;
```

Comment:
Every client will have to do the same thing in every lang. I think we should consider changing this settings to be per client on the server end.